src/test/java/org/broadinstitute/hellbender/utils/pileup/ReadPileupUnitTest.java

```diff
@@ -75,6 +75,58 @@ public void testSplitByReadGroup() {
+        int s = 0;
+        for(final String sample: Arrays.asList("sample1", "sample1", "sample2", null)) {
+            readGroups[s] = new SAMReadGroupRecord("rg"+s);
+            readGroups[s++].setSample(sample);
+        }
+        header.setReadGroups(Arrays.asList(readGroups));
+
+        // reads
+        final int rgCoverage = 4;
+        final List<GATKRead> reads = new ArrayList<>(rgCoverage*readGroups.length);
+        for(int i = 0; i < rgCoverage; i++) {
+            for(final SAMReadGroupRecord rg: readGroups) {
+                final GATKRead r = ArtificialReadUtils.createArtificialRead(header, rg.getReadGroupId() + "_" + rg.getSample() + "_" + i, 0, 1, 10);
+                r.setReadGroup(rg.getId());
+                reads.add(r);
+            }
+        }
+
+        // pileup
+        final ReadPileup pileup = new ReadPileup(loc, reads, 1);
+        // sample1
+        final ReadPileup pileupSample1 = pileup.getPileupForSample("sample1", header);
+        Assert.assertEquals(pileupSample1.size(), rgCoverage*2, "Wrong number of elements for sample1");
+        Assert.assertTrue( pileupSample1.getReadGroupIDs().contains("rg0"), "Pileup for sample1 should contain rg0");
+        Assert.assertTrue( pileupSample1.getReadGroupIDs().contains("rg1"), "Pileup for sample1 should contain rg1");
+        Assert.assertFalse(pileupSample1.getReadGroupIDs().contains("rg2"), "Pileup for sample1 shouldn't contain rg2");
+        Assert.assertFalse(pileupSample1.getReadGroupIDs().contains("rg3"), "Pileup for sample1 shouldn't contain rg3");
+
+        // sample2
+        final ReadPileup pileupSample2 = pileup.getPileupForSample("sample2", header);
+        Assert.assertEquals(pileupSample2.size(), rgCoverage, "Wrong number of elements for sample2");
+        Assert.assertFalse(pileupSample2.getReadGroupIDs().contains("rg0"), "Pileup for sample2 shouldn't contain rg0");
+        Assert.assertFalse(pileupSample2.getReadGroupIDs().contains("rg1"), "Pileup for sample2 shouldn't contain rg1");
+        Assert.assertTrue( pileupSample2.getReadGroupIDs().contains("rg2"), "Pileup for sample2 should contain rg2");
+        Assert.assertFalse(pileupSample2.getReadGroupIDs().contains("rg3"), "Pileup for sample2 shouldn't contain rg3");
+
+        // null sample
+        final ReadPileup pileupNull = pileup.getPileupForSample(null, header);
+        Assert.assertEquals(pileupNull.size(), rgCoverage, "Wrong number of elements for null sample");
+        Assert.assertFalse(pileupNull.getReadGroupIDs().contains("rg0"), "Pileup for null sample shouldn't contain rg0");
+        Assert.assertFalse(pileupNull.getReadGroupIDs().contains("rg1"), "Pileup for null sample shouldn't contain rg1");
+        Assert.assertFalse(pileupNull.getReadGroupIDs().contains("rg2"), "Pileup for null sample shouldn't contain rg2");
+        Assert.assertTrue( pileupNull.getReadGroupIDs().contains("rg3"), "Pileup for null sample should contain rg3");
```

Comment:
hmm, this seems to conflict with the doc you added. The doc in getPileupForSample says that passing null sample will get me reads without read groups. But this test here clearly shows that the readgroups for those reads are set to rg3. Can you clarify+fix?